libs/IMUKit/source/IMUKit.cpp

```diff
@@ -26,18 +70,45 @@ void IMUKit::stop()
 
 void IMUKit::setOrigin()
 {
-	_mahony.setOrigin();
+	// TODO(@ladislas): to implement
+	// ? Reseting the algorithm might not be the best answer as it takes a few second to stabilize
+	// ? Because the readings are very stable, it would be easier to take the current yaw
+	// ? and start counting from there
 }
 
-auto IMUKit::getAngles() -> std::array<float, 3>
+auto IMUKit::getEulerAngles() const -> EulerAngles
 {
-	return {_mahony.getPitch(), _mahony.getRoll(), _mahony.getYaw()};
+	return _euler_angles;
 }
 
-void IMUKit::computeAngles(const interface::LSM6DSOX::SensorData &imu_data)
+void IMUKit::drdy_callback(const interface::LSM6DSOX::SensorData data)
 {
-	auto xl_data			= std::make_tuple(imu_data.xl.x, imu_data.xl.y, imu_data.xl.z);
-	auto gy_data			= std::make_tuple(imu_data.gy.x, imu_data.gy.y, imu_data.gy.z);
-	constexpr auto mag_data = std::make_tuple(0.0F, 0.0F, 0.0F);
-	_mahony.update(xl_data, gy_data, mag_data);
-}
+	// ? Note: For a detailed explanation on the code below, checkout
+	// ? https://github.com/leka/LekaOS/tree/develop/spikes/lk_sensors_imu_lsm6dsox_fusion_calibration
```

Comment:
I like this comment to keep trace of "original" or "inspiration" code, I suggest to also add a tag to the lib under the fusion's include as you added for the spike.

https://github.com/leka/LekaOS/blob/41f3e6c652fc5740d3dfe61fc7b92270bd6612f9/spikes/lk_sensors_imu_lsm6dsox_fusion_calibration/main.cpp#L14-L16